custom_components/tuya_local/devices/aspen_fan.yaml

```diff
@@ -0,0 +1,95 @@
+name: Aspen Fan
+primary_entity:
+  entity: fan
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+    - id: 2
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: in
+          value: cool
+        - dps_val: out
+          value: exhaust
+        - dps_val: exch
+          value: exchange
+    - id: 3
+      type: integer
+      name: speed
+      mapping:
+        - dps_val: 1
+          value: low
+        - dps_val: 2
+          value: medium
+        - dps_val: 3
+          value: high
+secondary_entities:
+  - entity: climate
+    name: Bedroom Fan
+    dps:
+      - id: 1
+        type: boolean
+        name: switch
+      - id: 2
+        type: string
+        name: preset_mode
+        mapping:
+          - dps_val: in
+            value: intake
+          - dps_val: out
+            value: exhaust
+          - dps_val: exch
+            value: exchange
+      - id: 101
+        type: boolean
+        mapping:
```

Comment:
Should it be "off" and "fan_only" rather than "fan_only" and "cool"?  Or is this an evaporative cooler or something?  In that case, climate as primary entity would be better than fan.

Reply:
It looks like this should interact with dps 1, so add `hidden: true` to dps 1, and here, add
```
    constraint: switch
    conditions:
       - dps_val: false
         value: "off"
       - dps_val: true
          value: fan_only
```
for both the fan_only and cool mappings.

Reply:
> 

id: 101, this controls the "temperature/climate set point function. It has two modes, the way I have them named right now:
fan_only: the fan runs regardless of the room temperature
cool: works like the 'cool' function on an AC, as the room reaches the set temperature the fan will turn off, and turn back on if the room temperature rises. I can rename the values but I was worried "off" may be confusing as this does not turn off the fan. This is also the reason I had added it as a climate device and a fan device, this way it was easier to integrate with climate lovelace cards. Let me know what you think the best way to do this.

Reply:
I think because of the thermostat feature, it is a better fit with the climate entity rather than fan. But I think it is the switch (dps 1) that is HVAC mode, with off and fan_only as values. Dps 101 should be preset mode, I think, selecting between "constant" and "auto" (or similar names), with temperature setting invalid when it is false.
The fan direction should be a separate select entity I think.